google/cloud/bigquery/v2/minimal/internal/job_logging_test.cc

```diff
@@ -0,0 +1,170 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigquery/v2/minimal/internal/job_logging.h"
+#include "google/cloud/bigquery/v2/minimal/internal/job_rest_stub.h"
+#include "google/cloud/bigquery/v2/minimal/testing/mock_job_log_backend.h"
+#include "google/cloud/bigquery/v2/minimal/testing/mock_job_rest_stub.h"
+#include "google/cloud/common_options.h"
+#include "google/cloud/log.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+using ::google::cloud::bigquery_v2_minimal_testing::MockBigQueryJobRestStub;
+using ::testing::HasSubstr;
+using ::testing::IsEmpty;
+
+std::shared_ptr<BigQueryJobLogging> CreateMockJobLogging(
+    std::shared_ptr<BigQueryJobRestStub> mock) {
+  return std::make_shared<BigQueryJobLogging>(std::move(mock), TracingOptions{},
+                                              std::set<std::string>{});
```

Comment:
These `components` appear to be unused.  Should we get rid of them until they're needed?